framework/wazuh/core/config/models/server.py

```diff
@@ -0,0 +1,333 @@
+        Whether to remove subdirectories if they are empty.
+    extra_valid : bool
+        Whether to perform extra validation.
+    """
+    dir: str
+    description: str
+    permissions: PositiveInt
+    source: str
+    names: List[str]
+    recursive: bool
+    restart: bool
+    remove_subdirs_if_empty: bool
+    extra_valid: bool
+
+
+class ServerInternalConfig(BaseModel):
+    """Configuration for server internal settings.
+
+    Parameters
+    ----------
+    files : List[SharedFiles]
+        List of shared file configurations.
+    excluded_files : List[str]
+        List of files to be excluded.
+    excluded_extensions : List[str]
+        List of file extensions to be excluded.
+    """
+    files: List[SharedFiles]
+    excluded_files: List[str]
+    excluded_extensions: List[str]
+
+    def get_dir_config(self, name: str) -> Optional[SharedFiles]:
+        """Retrieve the shared file configuration for a given directory name.
+
+        Parameters
+        ----------
+        name : str
+            The name of the directory to look up.
+
+        Returns
+        -------
+        Optional[SharedFiles]
+            The configuration for the shared files in the specified directory,
+            or None if not found.
+        """
+        for file in self.files:
+            if file.dir == name:
+                return file
+
+        return None
+
+
+class CTIConfig(BaseModel):
+    """Configuration for CTI settings.
+
+    Parameters
+    ----------
+    update_check : bool
+        Whether to perform an update check. Default is True.
+    url : str
+        The URL for the CTI service. Default is "https://cti.wazuh.com".
+    """
+    update_check: bool = True
+    url: str = "https://cti.wazuh.com"
+
+
+class ServerConfig(BaseModel):
+    """Configuration for the server.
+
+    Parameters
+    ----------
+    port : PositiveInt
+        The port on which the server will listen. Default is 1516.
+    bind_addr : str
+        The address to bind to. Default is "0.0.0.0".
+    nodes : List[str]
+        List of nodes in the server.
+    hidden : bool
+        Whether the server is hidden. Default is False.
+    update_check : bool
+        Whether to perform an update check. Default is False.
+    node : NodeConfig
+        Configuration for the server node.
+    worker : WorkerConfig
+        Configuration for worker nodes. Default is WorkerConfig().
+    master : MasterConfig
+        Configuration for the master node. Default is MasterConfig().
+    communications : CommunicationsConfig
+        Configuration for communications. Default is CommunicationsConfig().
+    logging : LoggingConfig
+        Logging configuration. Default is LoggingConfig(level="debug2").
+    cti : CTIConfig
+        Configuration for CTI settings. Default is CTIConfig().
+    _internal : ServerInternalConfig
+        Internal server configurations. These settings are internal
+    """
+    port: PositiveInt = 1516
+    bind_addr: str = "0.0.0.0"
```

Comment:
We should bind to `localhost` by default

Reply:
Fixed in 74eb07d1ae7f6a5a3ef80c50ff5984cb3af28939.